File: app/src/main/java/com/demo/justapp/exchanger/di/data/DataComponent.java
package com.demo.justapp.exchanger.di.data;

import com.demo.justapp.exchanger.di.scope.Data;
import com.demo.justapp.exchanger.presentation.rates.activity.RatesActivity;
import com.demo.justapp.exchanger.presentation.rates.fragment.RatesFragment;

import dagger.Component;
import dagger.Subcomponent;

/**
 * Компонент для работы после авторизации {@link Component} в приложении.
 *
 * @author Sergey Rodionov
 */
@Data
@Subcomponent(modules = DataModule.class)
public interface DataComponent {

    @Subcomponent.Builder
    interface Builder {
        DataComponent build();
    }

    void inject(RatesActivity activity);

    void inject(RatesFragment fragment);

}

File: app/src/main/kotlin/com/demo/justapp/exchanger/di/data/DataComponent.kt
package com.demo.justapp.exchanger.di.data

import com.demo.justapp.exchanger.di.scope.Data
import com.demo.justapp.exchanger.presentation.rates.activity.RatesActivity
import com.demo.justapp.exchanger.presentation.rates.fragment.RatesFragment
import dagger.Subcomponent

/**
 * Компонент для работы после авторизации [dagger.Component] в приложении.
 *
 * @author Sergey Rodionov
 */
@Data
@Subcomponent(modules = [DataModule::class])
interface DataComponent {

    @Subcomponent.Builder
    interface Builder {
        fun build(): DataComponent
    }

    fun inject(activity: RatesActivity)
    fun inject(fragment: RatesFragment)
}